Marlin/src/module/temperature.cpp

```diff
@@ -64,29 +64,27 @@ Temperature thermalManager;
  * dead-code elimination will (hopefully) optimize the TEMP_ERROR_PSTR macro
  * to use the least number of conditionals for the value of EXTRUDERS
  */
-#if not defined(MSG_HEATING_FAILED_LCD_BED)
+#ifndef MSG_HEATING_FAILED_LCD_BED
   #define MSG_HEATING_FAILED_LCD_BED MSG_HEATING_FAILED_LCD " " MSG_HEATER_BED
 #endif
 
-#define CONCAT_PSTR(a,b) a " " b
-
 #if HAS_TEMP_BED
   #define TEMP_ERROR_PSTR(msg, e) \
       ((e == -1) ? msg ## _BED : \
-        (((EXTRUDERS > 0) && (e == 0)) ? CONCAT_PSTR(msg, MSG_E1) : \
-          (((EXTRUDERS > 1) && (e == 1)) ? CONCAT_PSTR(msg, MSG_E2) : \
-            (((EXTRUDERS > 2) && (e == 2)) ? CONCAT_PSTR(msg, MSG_E3) : \
-              (((EXTRUDERS > 3) && (e == 3)) ? CONCAT_PSTR(msg, MSG_E4) : \
-                (((EXTRUDERS > 4) && (e == 4)) ? CONCAT_PSTR(msg, MSG_E5) : \
-                  CONCAT_PSTR(msg, MSG_E5 "+")))))))
+        (((EXTRUDERS > 0) && (e == 0)) ? _CAT(msg, MSG_E1) : \
```

Comment:
Meow?

Reply:
A popular macro, apparently, `_CAT` takes an arbitrary number of atoms and combines them. So actually, now it's…
```cpp
_CAT(MSG_E1, " ", msg)
```

Reply:
Nice. I found the definition of the macro:

`#define _CAT(a, ...) a \#\# \_\_VA_ARGS\_\_`

And I learned two things:
1. \#\# works for strings
2. The preprocessor accepts variable arguments.